src/libstore/derivations.cc

```diff
@@ -1210,72 +1210,197 @@ std::optional<BasicDerivation> Derivation::tryResolve(
     return resolved;
 }
 
+/**
+ * Process `InputAddressed`, `Deferred`, and `CAFixed` outputs.
+ *
+ * For `InputAddressed` outputs or `Deferred` outputs:
+ *
+ * - with `Regular` hash kind, validate `InputAddressed` outputs have the
+ *   correct path (throws if mismatch), invoke `deferredCallback` for
+ *   `Deferred` outputs with the computed output path, then invoke
+ *   `envCallback` with the path.
+ *
+ * - with `Deferred` hash kind, validate that the output is either
+ *   `InputAddressed` (error) or `Deferred` (correct).
+ *
+ * For `CAFixed` outputs, invoke `envCallback` with the computed path.
+ */
+static void processForOutputPaths(
```

Comment:
The docs and method name don't really tell me what this code is trying to accomplish at all.

It's computing output paths?